### ISSUE TITLE
Purge Database Size for event

### ISSUE THREAD

**Issue by hallard** [gotthardp/lorawan-server#606]
We need to decrease database size, on multitech GW with lot of devices, sometime admin interface does not answer anymore until we delete the following files: 

- `event.DLC` and `event.DCD`
- `rxframe.DLC` and `rxframe.DCD`
- `LATEST.LOG`

As asked in decrease database size #601 We understand we can use parameter 
```
 % amount of rxframes retained for each device/node
    {retained_rxframes, 50},
```
Is there something similar for events ?

Thansks

**Comment by gotthardp** [comment]
Yes, there is: `{event_lifetime, 86400}`. The number means how many seconds is an event displayed before it gets deleted.

**Comment by hallard** [comment]
Looks like #601 I setup this parameter to 3600 and still have events from 2hours, could it be a localtime or timezone issue?
I've purged both from web interface. Can we purge thru API until it's fixed?

**Comment by gotthardp** [comment]
this is really weird, will have a look

the API allows you to delete all entries in a table (that's what the purge buttons do). simply `HTTP DELETE /api/events`, but be careful-- it deletes everything without any warning and any way to recover (that's why its not documented)

**Comment by autodomain** [comment]
:(
Il mer 27 mar 2019, 16:10 Petr Gotthard <notifications@github.com> ha
scritto:

**Comment by gotthardp** [comment]
@hallard To further debug this issue I made the following updates to the latest master:
 * I added new config parameter `{trim_interval, 3600}` to define how often (in seconds) is the database trimmed. Note that if you say {event_lifetime, 3600} and {trim_interval, 3600} your events will never be older than 2 (two!) hours.
 * Upon startup the system (debug) prints `Using config:` following a list of config parameters to make sure nothing was misspelled not misplaced.

**Comment by hallard** [comment]
@gotthardp 
Thanks, I'll try this one setting bot to 1800 to see, but not today, we need to finalise production by the end of day

**Comment by hallard** [comment]
@gotthardp 

Looks like still not working for RX frames
````
    % amount of rxframes retained for each device/node
    {retained_rxframes, 10},
    % events duration on interface
    {event_lifetime, 1800},
    {trim_interval, 1800},

````

Frames pages 458 frames
![image](https://user-images.githubusercontent.com/2471931/55237459-82559500-5232-11e9-9406-1e710b86b50c.png)

Most recent before
![image](https://user-images.githubusercontent.com/2471931/55237331-386caf00-5232-11e9-9ef0-577c2fdbb647.png)

Older before (2 days ago)
![image](https://user-images.githubusercontent.com/2471931/55237422-710c8880-5232-11e9-9c05-7c0023c056b6.png)

Events looks better despite the fact there are 3 old events but all other are fine
![image](https://user-images.githubusercontent.com/2471931/55237553-b761e780-5232-11e9-8182-101fb83c3296.png)

**Comment by gotthardp** [comment]
Concerning the events: please see the *Last Occurred* date. There are no old events.

**Comment by gotthardp** [comment]
Concerning the Frames: what you posted is not necessarily wrong, could you please (wait 1800 seconds after server start and then) filter out only one DevAddr? Is there a DevAddr that has more than a high number of old frames or not?

**Comment by jot4p** [comment]
Hi, 

It's possible to remove all received frames, removing rxframe.DLC and rxframe.DCD ? 
Can I remove this with loraserver running ?? 

What is the best procedure ? Because my rxframes page does not load......

**Comment by JoobyPM** [comment]
Dear @gotthardp is it safe to remove rxframe.DLC and rxframe.DCD ?
Or to configure retained_rxframes=0?
Thank you for your reply.
